scripts/azureml-assets/azureml/assets/environment/pin_image_versions.py

```diff
@@ -11,11 +11,27 @@
 
 from azureml.assets.util import logger
 
+from tenacity import retry, stop_after_attempt, wait_fixed
+
 LATEST_TAG = "latest"
 # Handles registry/image_name:{{latest-image-tag}} and registry/image_name:{{latest-image-tag:regex}}
 LATEST_IMAGE_TAG = re.compile(r"([^\"'\s]+):\{\{latest-image-tag(?::(.+))?\}\}")
 
 
+def log_attempt(retry_state):
+    logger.log_debug(f"Unable to open url, retrying image manifest call... (Attempt: {retry_state.attempt_number}/3)")

```

Comment:
Now that you've verified that retries are happening, do you think it's useful to continue logging retries? I'm on the fence about it, largely because we'll likely need to add this for PyPI calls too and would like a simple pattern we can replicate.